### DISCUSSION TITLE
feat: do not autoload if neovim was started with piped data

### DISCUSSION THREAD

**Discussion by WieeRd** [olimorris/persisted.nvim#140] [Ideas]
I think autoloading should be disabled if Neovim was started with piped data, e.g. `ls -l | nvim`.
It is very unlikely that the user wants the session to be autoloaded in this situation.
Just like when explicitly passing in the files to open via command line arguments.

However, I have no idea how to detect if Neovim was started this way...
If anyone know APIs related to this, give me a heads up.

**Comment by olimorris** [comment]
The new rewrite of the plugin allows you to customise how autoloading works so you can build this into your config.

**Reply by WieeRd** [reply]
Not without figuring out how to detect the piped data in the first place, which was the main concern of this question.

**Reply by olimorris** [reply]
I'd ask this on the Neovim discourse
